CONTRIBUTING.md

```diff
@@ -1,9 +1,203 @@
+  * [Enhancements vs bugs](#enhancements-vs-bugs)
+  * [Workflow](#workflow)
+    * [Forking and cloning this repository](#forking-and-cloning-this-repository)
+    * [Issues solving](#issues-solving)
+    * [Code review](#code-review)
+  * [Issues and bug reporting](#issues-and-bug-reporting)
+* [How to contribute](#how-to-contribute)
+  * [Simple workflow for bugifx](#simple-workflow-for-bugifx)
+    * [Branch naming conventions](#branch-naming-conventions)
+  * [Coding style and coding rules](#coding-style-and-coding-rules)
+    * [Code Like a Pythonista: Idiomatic Python](#code-like-a-pythonista-idiomatic-python)
+    * [Coding Style: Readability Counts](#coding-style-readability-counts)
+    * [Style Guide for Python Code](#style-guide-for-python-code)
+    * [Whitespace](#whitespace)
+    * [Naming](#naming)
+* [Testing and code coverage](#testing-and-code-coverage)
+* [Code coverage](#code-coverage)
+* [Keep your GitHub fork updated](#keep-your-github-fork-updated)
+    * [Tracking changes](#tracking-changes)
+    * [Verify with:](#verify-with)
+    * [Updating](#updating)
+* [Final notes](#final-notes)
 
-For more details please see our general [CONTRIBUTING.md](https://github.com/ARMmbed/greentea/blob/master/docs/CONTRIBUTING.md) document.
+# Contributing
+We really appreciate your contributions! We are an open source project and we need your help. We want to keep it as easy as possible to contribute changes that get things working in your environment. There are a few guidelines that we need contributors to follow so that we can have a chance of keeping on top of things.
 
-# Contributor agreement
+## Contributor agreement
 For your pull request to be accepted, we will need you to agree to our [contributor agreement](http://developer.mbed.org/contributor_agreement/) to give us the necessary rights to use and distribute your contributions. (To click through the agreement create an account on mbed.com and log in.)
 
-<3
+
+## Enhancements vs bugs
+Enhancements are:
+* New features implementation.
+* Code refactoring.
+* Coding rules or coding style improvements.
+* Code comments improvement.
+* Documentation work.
+Bugs are:
+* Issues raised internally or externally by application users.
+* Internally (from the ARM mbed team) created issues from the Continuous Integration pipeline and build servers.
+* Issues detected using automation tools such as compilers, sanitizers, static code and analysis tools.
+
+## Workflow
+### Forking and cloning this repository
+First [fork](https://help.github.com/articles/fork-a-repo/) this repository in GitHub, then clone it locally with:
+```
+$ git clone <repo-link>
+```
+Now you can create separate branches in the forked repository and prepare pull requests with changes.
+### Issues solving
+Simple work-flow issue solving process may contain below steps:
+1. Issue filed (by any user).
+2. Proper label assigned by gate-keeper.
+3. Bug-fixer forked and cloned.
```

Comment:
This line doesn't make much sense ? Is it telling people to fork this repo and then clone the fork locally?